include/cell/copy/shared_to_register.hpp

```diff
@@ -205,14 +205,118 @@ DEVICE static constexpr int col_exec_count() {
     return count;
 }
 
+template <typename Element>
+DEVICE void ldmatrix(const Element* src, Element* dst) {
+    uint32_t* reg = reinterpret_cast<uint32_t*>(dst);
+    uint32_t smem_addr = static_cast<uint32_t>(__cvta_generic_to_shared(src));
+    asm volatile(
+        "ldmatrix.sync.aligned.x4.m8n8.shared.b16 {%0, %1, %2, %3}, [%4];\n"
+        : "=r"(reg[0]), "=r"(reg[1]), "=r"(reg[2]), "=r"(reg[3])
+        : "r"(smem_addr));
+}
+
 }  // namespace
 
+namespace detail {
+
+template <typename Shared, const bool isRowMajor>
```

Comment:
Would using an enum instead of `isRowMajor` be a better choice?

Reply:
You're correct. This function can be typed by shared memory layout. 😭

Reply:
> Would using an enum instead of `isRowMajor` be a better choice?

I attempted various solutions, but I don't believe the current implementation is the most appropriate 😭 . The switch-case statement is somewhat inelegant. While functions cannot be partially specialized, converting these small functions into functors seems overly complex. I think a better solution can be developed in future PRs with improved organization of the Tile layout.